nihms-filesystem-builder/src/main/java/org/dataconservancy/nihms/builder/fs/FilesystemModelBuilder.java

```
@@ -18,17 +18,164 @@
+import org.dataconservancy.nihms.model.NihmsFile;
+import org.dataconservancy.nihms.model.NihmsMetadata;
 import org.dataconservancy.nihms.model.NihmsSubmission;
 
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.Properties;
+
 /**
  * Builds a NIHMS submission from a file on a locally mounted filesystem.  The file contains key-value pairs that would
  * loosely mimic the form data provided by the view layer.
+ *
+ * This implementation is for a bare-bones demo only - in particular we can have just one Person, and just one File
  */
 public class FilesystemModelBuilder implements SubmissionBuilder {
 
     @Override
     public NihmsSubmission build(String formDataUrl) throws InvalidModel {
-        return null;
+        Properties properties = new Properties();
+        InputStream is = null;
+
+        //The submission object to populate
+        NihmsSubmission submission = new NihmsSubmission();
+
+        //The file to add, and its list to add to submission
+        NihmsFile file = new NihmsFile();
+        List<NihmsFile> files = new ArrayList<>();
+
+        //The metadata object, and components to add to it ...
+        NihmsMetadata metadata = new NihmsMetadata();
+        NihmsMetadata.Journal journal = new NihmsMetadata.Journal();
+        NihmsMetadata.Manuscript manuscript = new NihmsMetadata.Manuscript();
+
+        //... including the person object and its list
+        NihmsMetadata.Person person = new NihmsMetadata.Person();
+        List<NihmsMetadata.Person> persons = new ArrayList<>();
+
+        try {
+            is = new FileInputStream(formDataUrl);
+            if (is == null) {
+                System.out.println("Sorry, unable to find submission properties file " + formDataUrl);
+                return null;
```

Instead of returning `null` I think you can throw `InvalidModel`